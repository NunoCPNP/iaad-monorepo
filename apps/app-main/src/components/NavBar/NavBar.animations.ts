export const animContainer = {
  hidden: { opacity: 1, scale: 0 },
  visible: {
    opacity: 1,
    scale: 1,
    transition: {
      staggerChildren: 0.1,
      delayChildren: 0.2,
    },
  },
}

export const animItem = {
  hidden: { y: 20, opacity: 0 },
  visible: {
    y: 0,
    opacity: 1,
  },
}
