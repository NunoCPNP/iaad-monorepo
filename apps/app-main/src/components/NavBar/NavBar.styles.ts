import styled from '@emotion/styled'

export const Wrapper = styled.nav`
  width: 100%;
  background-color: ${({ theme }) => theme.colors.accent};
  box-shadow: 0px 1px 8px 0px rgba(0, 0, 0, 0.3);
  transition: 0.3s ease background-color;
`

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  max-width: 140rem;
  height: 6rem;
  margin: 0 auto;

  @media only screen and (max-width: 1400px) {
    padding: 0 1rem;
  }
`

export const Items = styled.div`
  ul {
    display: flex;
  }

  li a {
    font-family: 'Roboto', sans-serif;
    text-transform: uppercase;
    margin-left: 1.5rem;
    font-size: 1.3rem;
    font-weight: 700;
    text-decoration: none;
    color: ${({ theme }) => theme.colors.secondary};
    transition: 0.2s ease-out color;

    :hover {
      color: ${({ theme }) => theme.colors.primary};
    }
  }

  @media only screen and (max-width: 768px) {
    display: none;
  }
`

export const Logo = styled.div`
  width: 4rem;
`
