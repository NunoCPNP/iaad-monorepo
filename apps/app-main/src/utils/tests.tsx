import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

const AllTheProviders = ({ children }) => {
  return <div>{children}</div>
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
