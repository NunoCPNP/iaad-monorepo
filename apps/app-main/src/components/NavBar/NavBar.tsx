import Link from 'next/link'
import Image from 'next/image'
import useTranslation from 'next-translate/useTranslation'
import { motion } from 'framer-motion'

import BurgerIcon from '../BurgerIcon'

import { Logo, Wrapper, Container, Items } from './NavBar.styles'
import { animContainer, animItem } from './NavBar.animations'

const navbaritems = [
  {
    label: 'home',
    link: '/',
    type: 'standard',
  },
  {
    label: 'who',
    link: '/#who',
    type: 'standard',
  },
  {
    label: 'services',
    link: '/#services',
    type: 'standard',
  },
  {
    label: 'contacts',
    link: '/#contacts',
    type: 'standard',
  },
  {
    label: 'ctc',
    link: '/#ctc',
    type: 'standard',
  },
]

const NavBar: React.FC = () => {
  const { t } = useTranslation()

  return (
    <Wrapper data-testid="navbar">
      <Container>
        <div>
          <Logo>
            <Image src="/iaad_logo.png" width={200} height={50} layout="fixed" />
          </Logo>
        </div>
        <Items>
          <motion.ul variants={animContainer} initial="hidden" animate={'visible'}>
            {navbaritems.map((link, index) => (
              <motion.li key={index} variants={animItem}>
                <Link href={link.link}>
                  <a>{t(`common:navbar.${link.label}`)}</a>
                </Link>
              </motion.li>
            ))}
          </motion.ul>
        </Items>
        <BurgerIcon />
      </Container>
    </Wrapper>
  )
}

export default NavBar
