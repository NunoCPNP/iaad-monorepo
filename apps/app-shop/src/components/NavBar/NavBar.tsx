import Link from 'next/link'
import { motion } from 'framer-motion'

import BurgerIcon from '../BurgerIcon'

import { Logo, Wrapper, Container, Items } from './NavBar.styles'

const container = {
  hidden: { opacity: 1, scale: 0 },
  visible: {
    opacity: 1,
    scale: 1,
    transition: {
      staggerChildren: 0.1,
      delayChildren: 0.2,
    },
  },
}

const item = {
  hidden: { y: 20, opacity: 0 },
  visible: {
    y: 0,
    opacity: 1,
  },
}

type Props = {
  navbar: any
}

const NavBar: React.FC<Props> = ({ navbar }) => {
  return (
    <Wrapper data-testid="navbar">
      <Container>
        <div className="navbar-logo">
          <Logo>{/* <img src="/logo192.png" alt="Website logo" /> */}</Logo>
        </div>
        <Items>
          <motion.ul variants={container} initial="hidden" animate={'visible'}>
            {navbar &&
              navbar.map((link) => (
                <motion.li key={link.id} variants={item}>
                  <Link href={link.link}>
                    <a>{link.label}</a>
                  </Link>
                </motion.li>
              ))}
          </motion.ul>
        </Items>
        <BurgerIcon />
      </Container>
    </Wrapper>
  )
}

export default NavBar
