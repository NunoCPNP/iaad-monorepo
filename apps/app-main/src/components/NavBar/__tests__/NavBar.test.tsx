import { render, cleanup } from '../../../utils/tests'

import NavBar from '../NavBar'

afterEach(cleanup)

describe('<NavBar /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<NavBar />)

    const navbar = queryAllByTestId(/navbar/i)
    expect(navbar).toHaveLength(1)
  })

  it('Should render BurgerIcon Component', () => {
    const { queryAllByTestId } = render(<NavBar />)

    const navbar = queryAllByTestId(/burgericon/i)
    expect(navbar).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<NavBar />)

    expect(asFragment(<NavBar />)).toMatchSnapshot()
  })
})
