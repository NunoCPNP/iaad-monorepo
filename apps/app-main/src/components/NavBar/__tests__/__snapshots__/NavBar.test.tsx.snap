// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> spec Should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  background-color: var(--navbarBackground);
  box-shadow: 0px 1px 8px 0px rgba(0, 0, 0, 0.3);
  -webkit-transition: 0.3s ease background-color;
  transition: 0.3s ease background-color;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  max-width: 140rem;
  height: 6rem;
  margin: 0 auto;
}

@media only screen and (max-width: 1400px) {
  .emotion-1 {
    padding: 0 1rem;
  }
}

.emotion-2 {
  width: 4rem;
}

.emotion-3 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 li a {
  font-family: 'Roboto',sans-serif;
  text-transform: uppercase;
  margin-left: 1.5rem;
  font-size: 1.3rem;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--navbarColor);
  -webkit-transition: 0.2s ease-out color;
  transition: 0.2s ease-out color;
}

.emotion-3 li a:hover {
  color: var(--navbarColorHover);
}

@media only screen and (max-width: 768px) {
  .emotion-3 {
    display: none;
  }
}

.emotion-4 {
  font-size: 50%;
  width: 6em;
  height: 4.5em;
  position: relative;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
  cursor: pointer;
}

.emotion-4 span {
  display: block;
  position: absolute;
  height: 0.6em;
  width: 50%;
  background: var(--burgericon);
  opacity: 1;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
}

.emotion-4 span:nth-of-type(even) {
  left: 50%;
  border-radius: 0 0.9em 0.9em 0;
}

.emotion-4 span:nth-of-type(odd) {
  left: 0em;
  border-radius: 0.9em 0 0 0.9em;
}

.emotion-4 span:nth-of-type(1),
.emotion-4 span:nth-of-type(2) {
  top: 0em;
}

.emotion-4 span:nth-of-type(3),
.emotion-4 span:nth-of-type(4) {
  top: 1.8em;
}

.emotion-4 span:nth-of-type(5),
.emotion-4 span:nth-of-type(6) {
  top: 3.6em;
}

.emotion-4.open span:nth-of-type(1),
.emotion-4.open span:nth-of-type(6) {
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.emotion-4.open span:nth-of-type(2),
.emotion-4.open span:nth-of-type(5) {
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.emotion-4.open span:nth-of-type(1) {
  left: 0.5em;
  top: 0.7em;
}

.emotion-4.open span:nth-of-type(2) {
  left: calc(50% - 0.5em);
  top: 0.7em;
}

.emotion-4.open span:nth-of-type(3) {
  left: -50%;
  opacity: 0;
}

.emotion-4.open span:nth-of-type(4) {
  left: 100%;
  opacity: 0;
}

.emotion-4.open span:nth-of-type(5) {
  left: 0.5em;
  top: 2.9em;
}

.emotion-4.open span:nth-of-type(6) {
  left: calc(50% - 0.5em);
  top: 2.9em;
}

@media only screen and (min-width: 769px) {
  .emotion-4 {
    display: none;
  }
}

<div>
    <nav
      class="emotion-0"
      data-testid="navbar"
    >
      <div
        class="emotion-1"
      >
        <div>
          <div
            class="emotion-2"
          >
            <div
              style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 200px; height: 50px;"
            >
              <img
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </div>
        <div
          class="emotion-3"
        >
          <ul
            style="opacity: 1; transform: scale(0) translateZ(0);"
          >
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/"
              >
                common:navbar.home
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#who"
              >
                common:navbar.who
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#services"
              >
                common:navbar.services
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#contacts"
              >
                common:navbar.contacts
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#ctc"
              >
                common:navbar.ctc
              </a>
            </li>
          </ul>
        </div>
        <div
          class=" emotion-4"
          data-testid="burgericon"
        >
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
